src/utils/csv.ts

```diff
@@ -84,3 +87,65 @@ export const formatVariables = (row: string[], keys: string[], startIndex: numbe
     type: Array.isArray(value) ? 'array' : typeof value,
   }));
 };
+
+/**
+ * Generates a cartesian product of arrays.
+ * @param arrays The arrays to generate the product from.
+ * @returns The generated product.
+ */
+export const cartesianProduct = (arrays: any[][]): any[][] => {
+  return arrays.reduce((a, b) => a.flatMap((d) => b.map((e) => [...d, e])), [[]]);
+};
+
+/**
+ * Generates a cartesian product of arrays in a more memory-efficient way, with the ability to adjust the maximum number of combinations.
+ * @param arrays The arrays to generate the product from.
+ * @param limit The maximum number of combinations to generate.
+ * @returns The generated product.
+ */
+export const complexCartesianProduct = (arrays: any[][], limit: number = 3000): any[][] => {
```

Comment:
Should be able to use a generic type "<T>" instead of "any".  
```
export const complexCartesianProduct = <T>(arrays: T[][], limit: number = 3000): T[][] => {
  const result: T[][] = [];
```